regression/cbmc/dynamic-object-02/test-no-cp.desc

```diff
@@ -0,0 +1,9 @@
+FUTURE
+main.c
+--no-simplify --no-propagation
+^EXIT=0$
+^SIGNAL=0$
+--
+^warning: ignoring
+--
+Check that uninitialized pointers are not considered pointing to dynamic objects
```

Comment:
Is this what we want? Or should it be non-det whether this is true?  

In particular, is there a doc where all these expectations are laid out?

Reply:
I think I agree with @danielsn, though I'm not clear what I as a user am trying to get out asserting on this property. But in general, seems safest for it to fail if you assert either way (in my mental model - you're asserting on UB, so whatever you assert, it should be false)

Reply:
@thk123 cbmc doesn't necessarily behave like that on UB. In general, one needs the appropriate check flags to be enabled to catch certain cases. For example, signed integer overflow is UB, though one needs to enable --signed-overflow-check to catch that case. Otherwise, the values will just wrap around.

In general though, I think it's a good idea for operations to be nondet in such cases as far as possible. So yes, I've changed the test to check for both the negated and the non-negated condition, and have specified that both should fail in the test.desc file.